core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -42,20 +50,29 @@ public void createCatalog() {
     File warehouse = temp.toFile();
     Configuration conf = new Configuration();
 
-    JdbcCatalog backendCatalog = new JdbcCatalog();
-    backendCatalog.setConf(conf);
-    Map<String, String> backendCatalogProperties = ImmutableMap.of(
-        CatalogProperties.WAREHOUSE_LOCATION, warehouse.getAbsolutePath(),
-        CatalogProperties.URI, "jdbc:sqlite:file::memory:?ic" + UUID.randomUUID().toString().replace("-", ""),
-        JdbcCatalog.PROPERTY_PREFIX + "username", "user",
-        JdbcCatalog.PROPERTY_PREFIX + "password", "password");
-    backendCatalog.initialize("backend", backendCatalogProperties);
+    this.restCatalog = new RESTCatalog((config) -> {
+      JdbcCatalog backendCatalog = new JdbcCatalog();
+      backendCatalog.setConf(conf);
+      Map<String, String> backendCatalogProperties = ImmutableMap.of(
+          CatalogProperties.WAREHOUSE_LOCATION, warehouse.getAbsolutePath(),
+          CatalogProperties.URI, "jdbc:sqlite:file::memory:?ic" + UUID.randomUUID().toString().replace("-", ""),
+          JdbcCatalog.PROPERTY_PREFIX + "username", "user",
+          JdbcCatalog.PROPERTY_PREFIX + "password", "password");
```

Comment:
The `backendCatalog` creation needed to be moved into the constructor so that the single use connection that is opened and then closed to fetch the server config doesn't permanently close the underlying JDBC catalog connection.

This way, the lifecycle of the connection to the JDBC catalog mirrors that of the HTTP client.

Reply:
I'm a bit concerned about this because it creates an entirely new JDBC catalog. It works for tests now, but if we were to send slightly different config back when loading a table, the catalog would create a new connection and wipe the DB.

Could this be solved instead by skipping `catalog.close()` in the `RESTCatalogAdaptor.close` method? We could simply close the `backendCatalog` in an `@AfterEach` method.

Reply:
Yeah that would be fine. I had it that way at one point in time (but never pushed it), so let me get it back that way again.